crates/rolldown_resolver/src/resolver.rs

```diff
@@ -66,9 +66,12 @@ impl<F: FileSystem + Default> Resolver<F> {
     };
 
     let resolve_options_with_default_conditions = OxcResolverOptions {
-      tsconfig: raw_resolve.tsconfig_filename.map(|p| TsconfigOptions {
-        config_file: p.into(),
-        references: oxc_resolver::TsconfigReferences::Disabled,
+      tsconfig: raw_resolve.tsconfig_filename.map(|p| {
+        let path = PathBuf::from(&p);
+        TsconfigOptions {
+          config_file: if path.is_relative() { cwd.join(path) } else { path },
```

Comment:
Probably should check if the path exist and give a warning about if the file is not existed in rolldown's side.

Reply:
Maybe a bug on my end

```
#[derive(Debug, Clone)]
pub struct TsconfigOptions {
    /// Allows you to specify where to find the TypeScript configuration file.
    /// You may provide
    /// * a relative path to the configuration file. It will be resolved relative to cwd.
    /// * an absolute path to the configuration file.
    pub config_file: PathBuf,

    /// Support for Typescript Project References.
    pub references: TsconfigReferences,
}
```

Reply:
I think the behavior of joining path to absolute in rolldown is correctly, due to we support passing cwd as an argument, we need to use the user-passed cwd.

Reply:
so if you pass a relative path to the tsconfig, oxc_resolver will find (and be able to read - although maybe it's read via the filesystem trait.), and parse it.

but when it starts looking in parent directories, it's resolved `ResolvedPath` is still relative, and not absolute. this is the bit that causes issues.